invenio_beard/processing/profiles.py

```
@@ -0,0 +1,242 @@
+
+    :praram signatures_map:
+        A dictionary representing signatures with keys
+        being exactly the same as the values of uuid of each
+        signature.
+
+        Example:
+            signatures_map = {u'a4156520-4248-a57f':
+                                {'author_affiliation': u'MIT, Cambridge, CTP',
+                                 'author_name': u'Wang, Yi-Nan',
+                                 'publication_id': u'1337489',
+                                 'signature_id': u'a4156520-4248-a57f',
+                                 'author_recid': u'10123',
+                                 'author_claimed: False}
+                              }
+
+    :param uuids:
+        A list of not-claimed signatures to be clustered, ie.
+        assigned to one of the claimed ones.
+
+        Example:
+            signatures = ['6e6e8b99-ee08-43a2-a7c6-136a531a47cc',
+                          '4a964f85-762f-4bff-8472-2f47e5f3bb26']
+
+    :param claimed_uuids:
+        A list of claimed signatures.
+
+        Example:
+            claimed_signatures = ['d63537a8-1df4-4436-b5ed-224da5b5028c']
+
+    :return:
+        A dictionary of manually clustered signatures, where the keys
+        represent uuids of claimed signatures and values are the lists
+        of not-claimed ones.
+
+        Example:
+            {'d63537a8-1df4-4436-b5ed-224da5b5028c':
+                ['6e6e8b99-ee08-43a2-a7c6-136a531a47cc',
+                 '4a964f85-762f-4bff-8472-2f47e5f3bb26']}
+    """
+    # Speed-up hack.
+    global distance_estimator
+
+    # Prepare results dictionary.
+    results = {}
+    for claimed_uuid in claimed_uuids:
+        results[claimed_uuid] = []
+
+    # Load the linkage model.
+    distance_model = os.path.abspath(os.path.join(os.path.dirname(
+                     __file__), '..', 'utils/data/linkage.dat'))
+    distance_estimator = pickle.load(open(distance_model, 'rb'))
+
+    try:
```

No `try..except:pass`